planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/utils.cpp

```diff
@@ -287,6 +287,36 @@ bool isPathInLanelets(
   return true;
 }
 
+bool pathFootprintExceedsTargetLaneBound(
+  const CommonDataPtr & common_data_ptr, const PathWithLaneId & path, const VehicleInfo & ego_info,
+  const double margin)
+{
+  if (common_data_ptr->direction == Direction::NONE || path.points.empty()) {
+    return false;
+  }
+
+  const auto & target_lanes = common_data_ptr->lanes_ptr->target;
+  const bool is_left = common_data_ptr->direction == Direction::LEFT;
+
+  const auto combined_target_lane = lanelet::utils::combineLaneletsShape(target_lanes);
+
+  auto it = path.points.rbegin();
+  for (; it < path.points.rend(); ++it) {
```

Comment:
curious why you didn't add it in the for statement?

Reply:
just thought it might help with readability, but actually I realized we don't need to use reverse iterator, so will change it to use range-based for